include/reporters/catch_reporter_junit.hpp

```diff
@@ -128,6 +128,18 @@ namespace Catch {
             xml.scopedElement( "system-err" ).writeText( trim( stdErrForSuite.str() ), false );
         }
 
+        static std::string fileNameTag( const std::set<std::string> &tags ) {
+            for( std::set<std::string>::const_iterator
+                 it = tags.begin(),
+                 itEnd = tags.end();
+                 it != itEnd;
+                 ++it ) {
+                if( !it->empty() && (*it)[0] == '#' )
```

Comment:
The performance difference is unlikely to matter, but I would prefer this to be done via `std::set::lower_bound`.